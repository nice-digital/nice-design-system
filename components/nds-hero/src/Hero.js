import React from "react";
import PropTypes from "prop-types";

import "../scss/hero.scss";

export const Hero = props => {
	const { actions, children, footer, header, intro, title } = props;
	return (
		<div className="hero">
			<div className="hero__container">
				{header && header}
				<div className="hero__body">
					<div className="hero__copy">
						<h1 className="hero__title">{title}</h1>
						{intro && <p className="hero__intro">{intro}</p>}
						{actions && <div className="hero__actions">{actions}</div>}
					</div>
					{children && <div className="hero__extra">{children}</div>}
				</div>
				{footer && <div className="hero__footer">{footer}</div>}
			</div>
		</div>
	);
};

Hero.propTypes = {
	actions: PropTypes.node,
	children: PropTypes.node,
	footer: PropTypes.node,
	header: PropTypes.node,
	intro: PropTypes.node,
	title: PropTypes.node.isRequired
};
