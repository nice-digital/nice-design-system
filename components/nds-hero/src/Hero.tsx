import React from "react";
import classnames from "classnames";
import "../scss/hero.scss";

export interface HeroProps {
	[prop: string]: unknown;
	header?: React.ReactNode;
	title: React.ReactNode;
	intro?: React.ReactNode;
	actions?: React.ReactNode;
	image?: string;
	isDark?: boolean;
	className?: string;
}

export const Hero: React.FC<HeroProps> = (props: HeroProps) => {
	const {
		actions,
		header,
		intro,
		title,
		image,
		isDark = false,
		className,
		...rest
	} = props;
	return (
		<div
			className={classnames(["hero", { "hero--dark": isDark }, className])}
			data-component="hero"
			{...rest}
		>
			<div className="hero__container">
				{header && header}
				<div className="hero__body">
					<h1 className="hero__title">{title}</h1>
					{intro && <p className="hero__intro">{intro}</p>}
					{actions && <div className="hero__actions">{actions}</div>}
				</div>
			</div>
			{image && (
				<div
					className="hero__image-container"
					style={{ backgroundImage: `url(${image})` }}
				></div>
			)}
		</div>
	);
};
