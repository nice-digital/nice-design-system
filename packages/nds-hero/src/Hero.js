import React from "react";
import PropTypes from "prop-types";

import "../scss/hero.scss";

export const Hero = props => {
	const { actions, children, footer, header, intro, title } = props;
	let Footer;
	if (footer && footer.content) {
		Footer = footer.elementType || "div";
	}
	return (
		<div className="hero">
			<div className="hero__container">
				{header && header}
				<div className="hero__body">
					<div className="hero__copy">
						<h1 className="hero__title">{title}</h1>
						{intro && <p className="hero__intro">{intro}</p>}
						{actions && <div className="hero__actions">{actions}</div>}
					</div>
					{children && <div className="hero__extra">{children}</div>}
				</div>
				{footer && <Footer className="hero__footer">{footer.content}</Footer>}
			</div>
		</div>
	);
};

Hero.propTypes = {
	actions: PropTypes.node,
	children: PropTypes.node,
	footer: PropTypes.shape({
		elementType: PropTypes.elementType,
		content: PropTypes.node.isRequired
	}),
	header: PropTypes.node,
	intro: PropTypes.node,
	title: PropTypes.node.isRequired
};
