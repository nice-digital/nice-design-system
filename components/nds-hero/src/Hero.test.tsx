import React from "react";
import { render } from "@testing-library/react";

import { Hero } from "./Hero";

const actions = (
	<>
		<a href="page-one">Go to page one</a>
		<a href="page-two">Go to page two</a>
	</>
);

const imageUrl = "https://placekitten.com/200/140";

const header = <p>Header content</p>;

describe("Hero", () => {
	it("should match the snapshot with props", () => {
		const wrapper = render(
			<Hero
				title="Welcoming title"
				intro="Introduction text"
				description="Description text"
				actions={actions}
				header={header}
				image={imageUrl}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it("should match the dark variant snapshot with props", () => {
		const wrapper = render(
			<Hero
				title="Welcoming title"
				intro="Introduction text"
				description="Description text"
				actions={actions}
				header={header}
				image={imageUrl}
				isDark={true}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it("should pass any number of child components via actions and extra props ", () => {
		const wrapper = render(
			<Hero
				title="Welcoming title"
				intro="Introduction text"
				description="Description text"
				actions={actions}
				data-track={false}
				image={imageUrl}
				className="mt--0"
			/>
		);
		const anchors = wrapper.getAllByRole("link");
		expect(anchors.length).toBe(2);
		expect(anchors[anchors.length - 1].textContent).toBe("Go to page two");
		expect(
			wrapper.container.querySelector("div.hero")?.getAttribute("data-track")
		).toBe("false");
		expect(wrapper.container.querySelector("div.hero")).toHaveClass("mt--0");
	});
});
