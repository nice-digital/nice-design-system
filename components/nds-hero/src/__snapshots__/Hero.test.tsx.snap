// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should match the dark variant snapshot with props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hero hero--dark"
        data-component="hero"
      >
        <div
          class="hero__container"
        >
          <p>
            Header content
          </p>
          <div
            class="hero__body"
          >
            <h1
              class="hero__title"
            >
              Welcoming title
            </h1>
            <p
              class="hero__intro"
            >
              Introduction text
            </p>
            <p
              class="hero__description"
            >
              Description text
            </p>
            <div
              class="hero__actions"
            >
              <a
                href="page-one"
              >
                Go to page one
              </a>
              <a
                href="page-two"
              >
                Go to page two
              </a>
            </div>
          </div>
        </div>
        <div
          class="hero__image-container"
          style="background-image: url(https://placekitten.com/200/140);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hero hero--dark"
      data-component="hero"
    >
      <div
        class="hero__container"
      >
        <p>
          Header content
        </p>
        <div
          class="hero__body"
        >
          <h1
            class="hero__title"
          >
            Welcoming title
          </h1>
          <p
            class="hero__intro"
          >
            Introduction text
          </p>
          <p
            class="hero__description"
          >
            Description text
          </p>
          <div
            class="hero__actions"
          >
            <a
              href="page-one"
            >
              Go to page one
            </a>
            <a
              href="page-two"
            >
              Go to page two
            </a>
          </div>
        </div>
      </div>
      <div
        class="hero__image-container"
        style="background-image: url(https://placekitten.com/200/140);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Hero should match the snapshot with props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hero"
        data-component="hero"
      >
        <div
          class="hero__container"
        >
          <p>
            Header content
          </p>
          <div
            class="hero__body"
          >
            <h1
              class="hero__title"
            >
              Welcoming title
            </h1>
            <p
              class="hero__intro"
            >
              Introduction text
            </p>
            <p
              class="hero__description"
            >
              Description text
            </p>
            <div
              class="hero__actions"
            >
              <a
                href="page-one"
              >
                Go to page one
              </a>
              <a
                href="page-two"
              >
                Go to page two
              </a>
            </div>
          </div>
        </div>
        <div
          class="hero__image-container"
          style="background-image: url(https://placekitten.com/200/140);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hero"
      data-component="hero"
    >
      <div
        class="hero__container"
      >
        <p>
          Header content
        </p>
        <div
          class="hero__body"
        >
          <h1
            class="hero__title"
          >
            Welcoming title
          </h1>
          <p
            class="hero__intro"
          >
            Introduction text
          </p>
          <p
            class="hero__description"
          >
            Description text
          </p>
          <div
            class="hero__actions"
          >
            <a
              href="page-one"
            >
              Go to page one
            </a>
            <a
              href="page-two"
            >
              Go to page two
            </a>
          </div>
        </div>
      </div>
      <div
        class="hero__image-container"
        style="background-image: url(https://placekitten.com/200/140);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
